src/ContainerInstance/custom/InvokeAzContainerInstanceCommand_ExecuteExpanded.cs

```diff
@@ -48,7 +48,7 @@ public partial class InvokeAzContainerInstanceCommand_ExecuteExpanded : global::
 
         private Task PullResponse()
         {
-            return Task.Factory.StartNew(async () =>
+            return Task.Run(async () =>
```

Comment:
https://code-maze.com/csharp-task-run-vs-task-factory-startnew/
Task.Factory.StartNew() vs Task.Run()
1. TaskCreationOptions.AttachedToParent vs TaskCreationOptions.DenyChildAttach? TaskCreationOptions.AttachedToParent as WriteObject() in PullResponse() depends on parent task.
2. Current vs Default TaskScheduler? Default TaskScheduler. No sure is current TaskScheduler same with default scheduler in ARM deployment environment?
3. Task<Task<string>> vs Task<string>?  No difference in this case due to no return.
4.  Object State? No difference in this case as no shared variable is scoped outside of the task block except socket.

Reply:
Maybe should use `Task.Factory.StartNew(action,  this._cancellationTokenSource.Token, TaskCreationOptions.None, TaskScheduler.Default);`? @andreygoran what's your thought?

Reply:
@BethanyZhou,

3. `Task<Task<void>>` vs `Task<void>` I think it's actually crucial in this case. We use `Task.WaitAll` later in code https://github.com/Azure/azure-powershell/blob/d6e1dc5e5759ae025a6509886ac5d7580113f5a6/src/ContainerInstance/custom/InvokeAzContainerInstanceCommand_ExecuteExpanded.cs#L45C13-L45C25 and `Task<Task<void>>` means that the waiting is over when the inner `Task` is returned, but this does not mean that the inner `Task` is actually completed by that time. However `Task.Run` does understand the semantics of `Task<Task<void>>` and actually waits until the inner task is completed.
Another solution would be to use `.Unwrap()` instead of switching `Task.Factory.StartNew` to `Task.Run`:
```
            return Task.Factory.StartNew(async () =>
            {
...
            }, this._cancellationTokenSource.Token).Unwrap();
```

> Maybe should use `Task.Factory.StartNew/Task.Run(action, this._cancellationTokenSource.Token, TaskCreationOptions.None, TaskScheduler.Default);`?

I randomly checked a few usages of `Task.Run` in the project, and I do not see any additional parameters used anywhere. So probably it's fine the way it is.

Reply:
Thanks for your opinions. 

I prefer to  
```
return Task.Factory.StartNew(async () =>
            {
...
            }, this._cancellationTokenSource.Token).Unwrap();
```

Because we need TaskCreationOptions.AttachedToParent but Task.Run() uses TaskCreationOptions.DenyChildAttach

Reply:
@BethanyZhou I added new commit that restores Task.Factory.StartNew and adds Unwrap()